sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -812,13 +812,6 @@ def test_neighbors_badargs():
                       cls, p=-1)
         assert_raises(ValueError,
                       cls, algorithm='blah')
-        nbrs = cls(algorithm='ball_tree', metric='haversine')
```

Comment:
I think it is implemented [L285](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neighbors/tests/test_ball_tree.py#L285) in ```test_ball_tree.py``` [L22](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neighbors/ball_tree.pyx#L22) in ```ball_tree.pyx```. The test  is to make sure haversine is not implemented for four ```neighbors``` classifiers and regressors.